### DISCUSSION TITLE
How to build a WASM binary for UIUA

### DISCUSSION THREAD

**Discussion by martin-saurer** [uiua-lang/uiua#576] [Interpreter Development]
Dear All,

May be this is a dumb question coming from a noob, but I think the UIUA Pad is based on UIUA WASM binary. Isn't it?
Is there any documentation on how to build a WASM binary for UIUA?

For J there is the "J Playground" which is based on a WASM binary for the J interpreter. This works pretty well on the iPad
using the a-shell App.

It would be great to have the same for UIUA on the iPad, that also works offline, like "J Playground".

Thanks in advance

Martin

**Comment by kaikalii** [comment]
This has been asked a good number of times, so I've added [a section](https://www.uiua.org/install#local-site) to the install page for how to do it.

**Reply by martin-saurer** [reply]
Thanks a lot! UIUA works like a charm on an iPad under a-Shell using the Python http.server.
